pkg/controller/statefulset/stateful_set.go

```diff
@@ -432,17 +432,17 @@ func (ssc *StatefulSetController) sync(key string) error {
 		return err
 	}
 
+	if err := ssc.adoptOrphanRevisions(set); err != nil {
+		return err
+	}
+
 	selector, err := metav1.LabelSelectorAsSelector(set.Spec.Selector)
 	if err != nil {
 		utilruntime.HandleError(fmt.Errorf("error converting StatefulSet %v selector: %v", key, err))
 		// This is a non-transient error, so don't retry.
 		return nil
 	}
 
-	if err := ssc.adoptOrphanRevisions(set); err != nil {
-		return err
-	}
-
```

Comment:
having second thoughts now, if the selector  cannot be formed without err, what is the point of adopting orphan revisions, so we should leave the order

Reply:
yes, you are right.but the selector seems to have been checked twice. once in adoptOrphanRevisions and another in here.

Reply:
i'll make another change.